### PR TITLE
VtoC: Prototype

### PR DESCRIPTION
**Pull request by kean**
Part of https://github.com/Automattic/wordpress-mobile/issues/102

To test: 

- Go through a happy path – ignore _everything_ 

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift [hunk 1]
```diff
@@ -33,11 +35,52 @@ extension MySiteViewController {
...
+            }
+        }
+        let view = VoiceToContentView(viewModel: viewModel)
+        let host = UIHostingController(rootView: view)
+
+        if UIDevice.isPad() {
+            host.modalPresentationStyle = .formSheet
+        } else {
+            if let sheetController = host.sheetPresentationController {
+                sheetController.detents = [.medium()]
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift:80]: On smaller devices, "Post from audio" is obscured in the sheet

iPhone SE (17.4.1) | iPhone 15 Pro (17.0) 
-- | --
 |

File: WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonCoordinator.swift [hunk 2]
```diff
@@ -24,8 +24,7 @@ import WordPressUI
     private let noticeAnimator = NoticeAnimator(duration: 0.5, springDampening: 0.7, springVelocity: 0.0)
 
     private func notice(for blog: Blog) -> Notice {
-        let showsStories = blog.supports(.stories)
-        let title = showsStories ? NSLocalizedString("Create a post, page, or story", comment: "The tooltip title for the Floating Create Button") : NSLocalizedString("Creates new post, or page", comment: " Accessibility hint for create floating action button")
+        let title = NSLocalizedString("Creates new post, or page", comment: " Accessibility hint for create floating action button")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonCoordinator.swift]: I wonder if we should use the same copy as showCreateButton - "Create a post or page"

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift [hunk 3]
```diff
@@ -33,11 +35,52 @@ extension MySiteViewController {
...
         let coordinator = CreateButtonCoordinator(self, actions: actions, source: source, blog: blog)
         return coordinator
     }
+
+    private func startPostFromAudioFlow() {
+        guard let blog else {
+            wpAssertionFailure("blog missing")
+            return
+        }
+        let viewModel = VoiceToContentViewModel(blog: blog) { [ weak self] transcription in
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift]: [nit]
```suggestion
        let viewModel = VoiceToContentViewModel(blog: blog) { [weak self] transcription in
```

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            }
+
+            VStack {
+                if let error = viewModel.errorMessage {
+                    // TODO: Add error handling
+                    Text(error)
+                        .foregroundStyle(.red)
+                } else {
+                    Text("File size limit: 25 MB")
+                    Text("Recording time limit: 5 minutes")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift:62]: Are these hardcoded values?

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    }
+}
+
+private struct VoiceToContentWelcomeView: View {
+    @ObservedObject fileprivate var viewModel: VoiceToContentViewModel
+
+    var body: some View {
+        VStack(spacing: 34) {
+            VStack(spacing: 16) {
+                Text("Post from Audio")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift:49]: Can you make the strings in this file localizable strings? Thanks!

**Review by momo-ozawa — COMMENTED** [review]
I understand that this is a prototype / a foundation to build on, so I haven't done a thorough test. The happy path seems to work well, but I noticed that if you record without saying anything, then tap "Done", the flow generates a paragraph block with the text "Thank you."

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift (hunk 5 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift:49]: On second thought, I can do this when I implement the updated design

File: WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonCoordinator.swift (hunk 2 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonCoordinator.swift]: Good point – I moved this to `Stings` too.

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift (hunk 3 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift]: Fixed

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift (hunk 4 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift:62]: Yeah, I added a todo in the latest PR.

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift:80]: I suggest addressing it when we start working on the final design.

**Comment by kean** [comment]
> if you record without saying anything, then tap "Done", the flow generates a paragraph block with the text "Thank you."

huh, good catch. I wonder if we can even do anything about it.

**Review by momo-ozawa — APPROVED** [review]

**Pull request merged**